docs/src/tutorials/simple_poisson_process.md

```diff
@@ -175,67 +175,101 @@ by adding or subtracting a constant vector from `u`.
-`VariableRateJump`
+upper-bound of ``2 λ``. We'll then re-encode the first (birth) jump as a
+`VariableRateJump`. Two types of `VariableRateJump`s are supported, general and
+bounded. The latter are generally more performant, but are also more restrictive
+in when they can be used. They also require specifying additional information
+beyond just `rate` and `affect!` functions.
+
+Let's see how to build a bounded `VariableRateJump` encoding our new birth
+process. We first specify the rate and affect functions, just like for a
+`ConstantRateJump`,
 ```@example tut1
 rate1(u,p,t) = p.λ * (sin(pi*t/2) + 1)
-lrate1(u,p,t) = p.λ
-urate1(u, p, t) = 2 * p.λ
-L1(u, p, t) = typemax(t)
 affect1!(integrator) = (integrator.u[1] += 1)
-vrj1 = VariableRateJump(rate1, affect1!; lrate=lrate1, urate=urate1, L=L1)
+```
+We next provide functions that determine a time interval over which the rate is
+bounded from above given `u`, `p` and `t`. From these we can construct the new
+bounded `VariableRateJump`:
+```@example tut1
+# We require that rate1(u,p,s) <= urate(u,p,s)
+# for t <= s <= t + rateinterval(u,p,t)
+rateinterval(u, p, t) = typemax(t)
+urate(u, p, t) = 2 * p.λ
+
+# Optionally, we can give a lower bound over the same interval.
+# This may boost computational performance.
+lrate(u, p, t) = p.λ
+
+# now we construct the bounded VariableRateJump
+vrj1 = VariableRateJump(rate1, affect1!; lrate, urate, rateinterval)
 ```
 
-Since births modify the population size `u[1]` and deaths `u[2]` occur at a rate
-proportional to the population size. We must represent this relation in
-a dependency graph. Note that the indices in the graph correspond to the order
-in which the jumps appear when the problem is constructed. The graph below
-indicates that births (event 1) modify deaths (event 2), but deaths do not
-modify births.
+Finally, to efficiently simulate the new jump process we must also specify a
+dependency graph. This indicates when a given jump occurs, which jumps in the
+system need to have their rates and/or rate bounds recalculated (for example,
+due to depending on changed components in `u`). We also assume the convention
+that a given jump depends on itself. Internally, JumpProcesses preserves the
+relative ordering of jumps of each distinct type, but always reorders all
+`ConstantRateJump`s to appear before any `VariableRateJump`s. As such, the
+`ConstantRateJump` representing the death process will have internal index 1,
+and our new bounded `VariableRateJump` for birth will have internal index 2.
+Since birth modifies the population size `u[1]`, and death occurs at a rate
+proportional to `u[1]`, when birth occurs we need to recalculate both of rates.
+In contrast, death does not change `u[1]`, and so when death occurs we only need
+to recalculate the death rate. The graph below encodes the dependents of the
+death (`dep_graph[1]`) and birth (`dep_graph[2]`) jumps respectively
 ```@example tut1
-dep_graph = [[2], []]
```

Comment:
Here I had omitted the self-loops since they are automatically included by the algorithm. My thought is that we might give the user the false impression that if he did not include the self-loops they would not get updated which is wrong.

Second, my understanding is that when jump `1` takes place then jumps `1` and `2` needs to be updated given that the dependencies come from `deps = p.dep_gr[p.next_jump]`. We have that `deathrate(u,p,t) = p.μ * u[1]`, so it should be updated when `p.next_jump == 1`. 

On the other hand, when jump `2` takes place the rate of `1` does not need an update.

Therefore, I derived `dep_grap = [[2], []]`. I think your description is also in agreement with this.

Reply:
Jump 1 is death as it is constant rate here, jump 2 is birth as it is variable rate.

Reply:
Ok, got it. Where I got confused was in `jprob = JumpProblem(dprob, Coevolve(), vrj1, deathcrj; dep_graph)`, since `vrj1` comes before `deathcrj`

But as we had discussed before and is somewhere else in the documentation, constant rate comes before variable rates in the dependency graph. Maybe worth pointing this out for the inattentive eye.

Reply:
Yup, the paragraph I added before the example explains this.